File: app/src/main/java/knf/kuma/videoservers/FenixServer.java
package knf.kuma.videoservers;

import android.content.Context;
import android.support.annotation.Nullable;

import org.json.JSONObject;
import org.jsoup.Jsoup;

import knf.kuma.commons.BypassUtil;
import knf.kuma.commons.PatternUtil;

import static knf.kuma.videoservers.VideoServer.Names.FENIX;

public class FenixServer extends Server {
    FenixServer(Context context, String baseLink) {
        super(context, baseLink);
    }

    @Override
    public boolean isValid() {
        return baseLink.contains("s=fenix");
    }

    @Override
    public String getName() {
        return FENIX;
    }

    @Nullable
    @Override
    public VideoServer getVideoServer() {
        try {
            String down_link = PatternUtil.extractLink(baseLink);
            String link = new JSONObject(Jsoup.connect(down_link.replace("embed", "check")).cookies(BypassUtil.getMapCookie(context)).userAgent(BypassUtil.userAgent).get().body().text()).getString("file");
            return new VideoServer(FENIX, new Option(getName(), null, link));
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: app/src/main/java/knf/kuma/videoservers/FenixServer.kt
package knf.kuma.videoservers

import android.content.Context
import knf.kuma.commons.PatternUtil
import knf.kuma.commons.jsoupCookies
import knf.kuma.videoservers.VideoServer.Names.FENIX
import org.json.JSONObject

class FenixServer internal constructor(context: Context, baseLink: String) : Server(context, baseLink) {

    override val isValid: Boolean
        get() = baseLink.contains("s=fenix")

    override val name: String
        get() = FENIX

    override val videoServer: VideoServer?
        get() {
            return try {
                val downLink = PatternUtil.extractLink(baseLink)
                val link = JSONObject(jsoupCookies(downLink.replace("embed", "check")).get().body().text()).getString("file")
                VideoServer(FENIX, Option(name, null, link))
            } catch (e: Exception) {
                e.printStackTrace()
                null
            }
        }
}
